h2o-automl/src/test/java/hex/ModelMojoWriterTest.java

```diff
@@ -0,0 +1,67 @@
+package hex;
+
+import ai.h2o.automl.targetencoding.BlendingParams;
+import ai.h2o.automl.targetencoding.TargetEncoder;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import water.Key;
+import water.TestUtil;
+import water.fvec.Frame;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.util.Map;
+
+/**
+ * This test should be moved to h2o-core module once we move all TargetEncoding related classes there as well. 
+ */
+public class ModelMojoWriterTest extends TestUtil implements ModelTestCommons {
+
+  @BeforeClass public static void stall() { stall_till_cloudsize(1); }
+  Frame trainFrame = parse_test_file("./smalldata/gbm_test/titanic.csv");
+  String fileName = "test_mojo_te.zip";
+  
+  private Map<String, Frame> getTEMap() {
+    
+    String responseColumnName = "survived";
+    asFactor(trainFrame, responseColumnName);
+    
+    BlendingParams params = new BlendingParams(3, 1);
+    String[] teColumns = {"home.dest", "embarked"};
+    TargetEncoder targetEncoder = new TargetEncoder(teColumns, params);
+    Map<String, Frame> testEncodingMap = targetEncoder.prepareEncodingMap(trainFrame, responseColumnName, null);
+    return testEncodingMap;
+  }
+
+  @Test public void writeModelToZipFile() {
+    TestModel.TestParam p = new TestModel.TestParam();
+    TestModel.TestOutput o = new TestModel.TestOutput();
+    o.setNames(trainFrame.names());
+    o._domains = trainFrame.domains();
+    
+    TestModel testModel = new TestModel(Key.make(),p,o);
+    Map<String, Frame> teMap = getTEMap();
+    testModel.addTargetEncodingMap(teMap);
+
+    String fileName = "test_mojo_te.zip";
+
+    try {
+      FileOutputStream modelOutput = new FileOutputStream(fileName);
+      testModel.getMojo().writeTo(modelOutput);
+      modelOutput.close();
+
+      Assert.assertTrue(new File(fileName).exists());
+
+    } catch (Exception ex) {
```

Comment:
why eat the exception? is it expected? if so, should be assert, otherwise let it fall though and fail the test

Reply:
@honzasterba again, this is because IDE will not let you compile as `new FileOutputStream()` and `modelOutput.close()` methods are declared as throwing exceptions. But you are right that I should have thrown something to fail the test. Simply forgot. There is actually a special method in JUnit that do this - it is called `org.junit.Assert.fail()` and I'm using it extensively in `TargetEncodingTest` for example. I'm using it when you expect test to fail in a particular way and you want access to a exception's message. 
```
@Test(expected = IllegalStateException.class)
```
does not give you this flexibility.

And I miss methods like `assertThrows()` from JUnit5 and `intercept` or `assertThrows` from `scalatest` library :(

Reply:
you can just declare the test mehtod with 'throws Exception'